### DISCUSSION TITLE
users-permissions/permissions endpoint is pointless

### DISCUSSION THREAD

**Discussion by abanchev** [strapi/strapi#6077] [Legacy Topics]
The current permissions endpoint is a bit pointless, it returns the names of all the permissions, however not the state (enabled or not) for the current role.
If we want to have the appropriate permissions we need to have the "getRole" permission, however that also returns users in the role, as well as rights for other roles.

i think  getPermissions(ctx) in UsersPermissions.js should check the current context and return the permissions for the requesting user.

**Comment by derrickmehaffy** [comment]
Hello,

I see you are wanting to ask a question that is not really a bug report or a feature request, questions should be directed to the [discussions](https://github.com/strapi/strapi/discussions)

I have converted this to a discussion for you, we try to keep Github clean and only for bug reports and feature requests.

**Reply by abanchev** [reply]
In issues if you click New Issue, and select Discussion from the list, it still creates an issue!

**Reply by derrickmehaffy** [reply]
I'm well aware of that, and there is a PR to remove it.

**Reply by derrickmehaffy** [reply]
https://github.com/strapi/strapi/pull/5844

**Comment by lauriejim** [comment]
Hello! This is related to an internal API.
I'm not sure to understand your point or what you are trying to do with it.

**Reply by abanchev** [reply]
I want to create an interface to edit content, and the visible options should be different depending on the user rights.

So I was hoping that the endpoint getPermissions would return the permissions that are applicable to the current user.

**Reply by derrickmehaffy** [reply]
You might be able to build a custom endpoint with something along the lines of the default `/users/me` but you should be able to run the GET /permissions using the role as a filter and look for enabled permissions. Let me do some testing.

Regardless for security you would want a custom endpoint anyway that takes into account the users JWT

**Reply by derrickmehaffy** [reply]
Yeah it doesn't look like this endpoint supports any kind of filtering (it doesn't even read the ctx). So this would need to be something you could build a custom endpoint for.

**Reply by abanchev** [reply]
I already wrote a custom endpoint, however it was confusing to have this endpoint listed in the roles and permissions editor, and it doesn't do what one would expect.
Maybe the roles that are for internal use should be hidden in the interface?
